crates/weaver_semconv/data/event.yaml

```diff
@@ -0,0 +1,181 @@
+groups:
```

Comment:
This file is not used in this PR, just including to show "why" I've added the Enum definition (even though we talked about excluding it).

Short Answer: is because the existing `device.app.lifecycle` has defined the usage of enum type in the `fields` 
BUT: the format is slightly different than currently defined for `AttributeType`

Reply:
Adding event makes sense to me, let's not add the deprecated "allow_custom_values" thing though